src/java.base/share/classes/java/util/BitSet.java

```diff
@@ -1398,4 +1401,48 @@ private int nextSetBit(int fromIndex, int toWordIndex) {
         }
     }
 
+    /**
+     * {@return a new {@link IntPredicate} representing the {@link BitSet#get(int)} method applied
+     * on an immutable snapshot of the current state of this BitSet}.
+     * <p>
+     * If the returned predicate is invoked with a {@code bitIndex} that is negative, the predicate
+     * will throw an IndexOutOfBoundsException just as the {@link BitSet#get(int)} method would.
+     * <p>
+     * Returned predicates are threadsafe and can be used without external synchronisation.  Furthermore,
+     * they are eligible for constant-folding optimization by the VM.
```

Comment:
>  Furthermore,
     * they are eligible for constant-folding optimization by the VM.

I'd probably skip this unless there's prior art to describe the constant-fold-eligibility of returned values from methods. :)